src/ParallelAlgorithms/NonlinearSolver/NewtonRaphson/LineSearch.cpp

```diff
@@ -0,0 +1,38 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#include <cmath>
+#include <cstddef>
+
+#include "Utilities/EqualWithinRoundoff.hpp"
+
+namespace NonlinearSolver::newton_raphson {
+
+double line_search(const size_t globalization_iteration_id,
+                   const double step_length, const double prev_step_length,
+                   const double residual, const double residual_slope,
+                   const double next_residual, const double prev_residual) {
+  if (globalization_iteration_id == 0) {
+    // First globalization step: minimum of quadratic interpolation
+    return -0.5 * square(step_length) * residual_slope /
+           (next_residual - residual - step_length * residual_slope);
```

Comment:
That's not an issue because this is the minimum of a quadratic polynomial with negative slope at x=0, so it's always positive. I added a bunch of asserts for the arguments though, e.g. that the slope is indeed negative.